File: source/CSharpWorks/VerifierTest.cs
using System;
using System.Threading.Tasks;
using VerifyXunit;
using Xunit;

namespace CSharpWorks
{
    [UsesVerify]
    public class VerifierTest
    {
        [Fact]
        public async Task Bar()
        {
            await Verifier.Verify("works");
        }
    }
}

File: source/VerifierPluginProblemRepro/VerifierTest.fs
[<VerifyXunit.UsesVerify>]
module VerifierPluginProblemRepro.VerifierTest

open VerifyXunit
open Xunit

let verify (value : 't) =
    Verifier.Verify<'t>(value).ToTask() |> Async.AwaitTask

[<Fact>]
let test () =
    verify "does not work"